addons/sale_stock/models/sale_order.py

```diff
@@ -533,6 +541,16 @@ def _get_outgoing_incoming_moves(self):
     def _get_procurement_group(self):
         return self.order_id.procurement_group_id
 
+    def _get_forecast_expected_date_per_move(self):
+        all_move_ids = {
+            move.id
+            for line in self
+            for move in line.move_ids
+            if line.state == 'sale' and move.product_id == line.product_id
+        }
+        all_moves = self.env['stock.move'].browse(all_move_ids)
+        return dict(all_moves.mapped(lambda m: (m.id, m.forecast_expected_date)))
+
```

Comment:
I think a new private method is not necessary here. You can inline the code inside the calling function instead. The calling function's code will remain readable and lean :slightly_smiling_face:

Reply:
Makes sense, will do 👍

Reply:
Amended it to do this